frontend/src/services/AuthService.ts

```diff
@@ -111,29 +118,44 @@ const parseToken = (authToken: CognitoUserSession): FamLoginUser => {
         authToken: authToken,
     };
     return famLoginUser;
-}
+};
 
 const removeFamUser = () => {
     storeFamUser(undefined);
     // clean up local storage for selected application
     localStorage.removeItem(CURRENT_SELECTED_APPLICATION_KEY);
-}
+};
 
 const storeFamUser = (famLoginUser: FamLoginUser | null | undefined) => {
     state.value.famLoginUser = famLoginUser;
     if (famLoginUser) {
-        localStorage.setItem(FAM_LOGIN_USER, JSON.stringify(famLoginUser));
+        // this call need auth token, so only call it when famLoginUser is not undefined
+        AdminMgmtApiService.adminUserAccessesApi
+            .adminUserAccessPrivilege()
+            .then(
+                (returnResult: AxiosResponse<AdminUserAccessResponse, any>) => {
+                    state.value.famLoginUser!.accesses =
+                        returnResult.data.access;
+                }
+            )
+            .catch((error: AxiosError) => {
+                setRouteToastError(error);
+            });
+        localStorage.setItem(
+            FAM_LOGIN_USER,
+            JSON.stringify(state.value.famLoginUser)
+        );
```

Comment:
I think, it is better off moving this retrieving user access logic within `storeFamUser` out to `AuthCallbackHandler.vue` or `AuthService.handlePostLogin()` (so the `storeFamUser` can only just be responsible for setting the cache). Maybe better in `handlePostLogin` function, since it really is meant for handling tasks, such as this, after user finally gets redirected back to app after Cognito.
For example, in `handlePostLogin` function, can do these steps:
```
 try {
       // DO await for : previous AWS stuff.
       // DO await for : previous refresh token stuff (this is when the app have the token ready)
       // DO await for: retrieving adminUserPrivilege() from backend.
       // DO: since the user privilege retrieved successfully at this point, we can set this into the famLoginUser. 
 }
 catch(error: any) {
        console.log('Not signed in');
        console.error('Error:', error);
        // DO: any other error handling you think it needs.
        return logout();
 }
```

This `handlePostLogin` was created at the early stage of this Vue project, the code looks old that it uses "Promise" style on return and error handling. Since it is an `async` function, it can ues "async-await` style and no need for `.then` (unless your feel having good reason to do so).

Reply:
Hi Ian, yeah I'm fine to move it up, but then after call the api, set the access in state famLoginUser, we need to set local storage again. That why initially I just put it inside the storeFamUser so it's set at one place. But I'm fine to move it up cause I feel maybe the logic will be more clear.